lib/generators/migration/migration_generator.rb

```diff
@@ -0,0 +1,56 @@
+require 'rails/generators/active_record/migration/migration_generator'
+
+# Overwrites the default `ActiveRecord::Generators::MigrationGenerator` to
+# include our own spec file for each migration generated.
+#
+module ManageIQ::Schema
+  class MigrationGenerator < ActiveRecord::Generators::MigrationGenerator
+    # Using both `source_root` (standard) and `source_paths` here so that we
+    # can default to the rails sources for the `migration.rb` template, but
+    # then use our own for the `migration_spec.rb` file.
+    source_root ActiveRecord::Generators::MigrationGenerator.source_root
+    source_paths << File.expand_path('templates', __dir__)
+
+    # Overwrite the defaulted generated .namespace method so that this class is
+    # found as the `migration` class.
+    #
+    # Some Thor/Rails magic happeninghere ... but basically it allows this to
+    # overwrite and work properly
+    #
+    def self.namespace(name = nil)
+      return super if name
+
+      "migration"
+    end
+
+    def self.miq_schema_root
+      File.expand_path(File.join("..", "..", ".."), __dir__)
```

Comment:
Would this be simpler?
```suggestion
      ManageIQ::Schema::Engine.root
```

Then you also have a Pathname object, so the other calls like db_migrate_path can piggyback on that

```ruby
def db_migrate_path
  miq_schema_root.join("db", "migrate")
end
```

Reply:
Done.